### PR TITLE
Solr reorganisation and adding of WMTS datatype to send to solr for f…

### PR DESCRIPTION
**Pull request by jmwenda**
…iltering

### CODE REVIEW

File: hypermap/aggregator/solr.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    def get_domain(url):
+        urlParts = urlparse(url)
+        hostname = urlParts.hostname
+        if hostname == "localhost":
+            return "Harvard"  # assumption
+        return hostname
+
+    @staticmethod
+    def is_solr_up():
+        solr_url = getattr(settings, 'SOLR_URL', 'http://localhost:8983/solr/geonode24')
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: not sure if this is a refuse

File: hypermap/aggregator/solr.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import math
+import json
+
+from urlparse import urlparse
+from django.conf import settings
+
+from aggregator.utils import mercator_to_llbbox
+
+
+class Solr_Hypermap(object):
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: would prefer SolrHypermap

File: hypermap/aggregator/solr.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                num = float(coord)
+                if (math.isnan(num)):
+                    return False
+                if (math.isinf(num)):
+                    return False
+            except ValueError:
+                return False
+        return True
+
+    solr_url = getattr(settings, 'SOLR_URL', 'http://localhost:8983/solr/geonode24')
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: another refuse?

File: hypermap/aggregator/solr.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                if (maxY > 90):
+                    maxY = 90
+                # ENVELOPE(minX, maxX, maxY, minY) per https://github.com/spatial4j/spatial4j/issues/36
+                wkt = "ENVELOPE({:f},{:f},{:f},{:f})".format(minX, maxX, maxY, minY)
+                # default for now as we do not have capabilitty to resolve vector/raster layers
+                dataType = "Polygon"
+                institution = "Harvard"
+                servicetype = layer.service.type
+                owsUrl = layer.service.url
+                if storeType == "remoteStore":
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: Refuse here as well

File: hypermap/aggregator/solr.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+                    institution = "Remote"
+                    if servicetype == "ESRI":
+                        dataType = "RESTServices"
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: why serviceteype and dataType are named differently?
and why their content is different?

File: hypermap/aggregator/solr.py [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+                    elif servicetype == "OGC:WMTS":
+                        dataType = "WMTSServices"
+                    else:
+                        dataType = "None"
+                domain = Solr_Hypermap.get_domain(owsUrl)
+                if (i == 0):
+                    i = layer.title
+                # we will need to resolve the projection issue at a later date
+                Solr_Hypermap.solr.add([{
+                                    "LayerId": "HyperMapLayer_" + str(i),
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: shouldn't we use a uuid here?

File: hypermap/aggregator/solr.py [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+                    maxX = 180
+                if (minY < -90):
+                    minY = -90
+                if (maxY > 90):
+                    maxY = 90
+                # ENVELOPE(minX, maxX, maxY, minY) per https://github.com/spatial4j/spatial4j/issues/36
+                wkt = "ENVELOPE({:f},{:f},{:f},{:f})".format(minX, maxX, maxY, minY)
+                # default for now as we do not have capabilitty to resolve vector/raster layers
+                dataType = "Polygon"
+                institution = "Harvard"
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: why do we need institution in Solr?

File: hypermap/aggregator/solr.py [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+                domain = Solr_Hypermap.get_domain(owsUrl)
+                if (i == 0):
+                    i = layer.title
+                # we will need to resolve the projection issue at a later date
+                Solr_Hypermap.solr.add([{
+                                    "LayerId": "HyperMapLayer_" + str(i),
+                                    "Name": layer.title,
+                                    "LayerDisplayName": layer.title,
+                                    "Institution": institution,
+                                    "Publisher": username,
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: I guess this will be harvested from metadata?

File: hypermap/aggregator/solr.py [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+                                    "MaxY": maxY,
+                                    "MaxX": maxX,
+                                    "CenterY": centerY,
+                                    "CenterX": centerX,
+                                    "HalfWidth": halfWidth,
+                                    "HalfHeight": halfHeight,
+                                    "Area": area,
+                                    "bbox_rpt": wkt}])
+                Solr_Hypermap.logger.error("solr record saved: " + layer.title)
+        except Exception as e:
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: I think celery will just interrupt here, so you can remove the error handling

File: hypermap/aggregator/solr.py [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+
+    @staticmethod
+    def clear_solr():
+        """Clear all indexes in the solr core"""
+        Solr_Hypermap.solr.delete(q='*:*')
+        print 'Solr core cleared'
+
+    @staticmethod
+    def solr_to_solr():
+        """create Solr records for layers in another Solr OGP instance"""
```
Comment (capooti) [inline hypermap/aggregator/solr.py]: what is it?

File: hypermap/aggregator/solr.py (hunk 5 above)
Comment (jmwenda) [inline hypermap/aggregator/solr.py]: Ben has insisted a while back on having both on the Solr side. i.e There was a period when he wanted to have different Imagery and Vector date from the same service type. i.e ESRI rest service could have vector/imagery. or polygon/line/point as datatypes to filter with. The other reason is from the OGP side, it was difficult to separate from ESRI WMS services from Non ESRI WMS services as both would be shown as WMS services. Please advise.

Comment (capooti) [inline hypermap/aggregator/solr.py]: as we are syncing with Solr, I would suggest that what in Django is:
layer.service.type = WMS/WMTS/ESRI/...
would be in Solr:
layer.service_type = WMS/WMTS/ESRI/...

If we need to store the geometric type information, we need to update the model in Django as well before we do it in Solr.
Same if we want to store the implementation of WMS (GeoServer/Esri/MapServer whatever), we need to add this information in HM, before doing it in Solr. But do we need it?

**Pull request merged**